### DISCUSSION TITLE
TSQLStatementReader.ParseStatements throws Object reference not set to an instance of an object. exception

### DISCUSSION THREAD

**Discussion by muneeebsheikh** [bruce-dunwiddie/tsql-parser#92] [Q&A] [answered]
I am trying to parse this select statement 

`SELECT system_user`

but the method TSQLStatementReader.ParseStatements() throws 

`Object reference not set to an instance of an object.`

stack trace 

```
at TSQL.Elements.Parsers.TSQLSelectColumnParser.Parse(ITSQLTokenizer tokenizer)
at TSQL.Clauses.Parsers.TSQLSelectClauseParser.Parse(ITSQLTokenizer tokenizer)
at TSQL.Statements.Parsers.TSQLSelectStatementParser.Parse()
at TSQL.Statements.Parsers.TSQLSelectStatementParser.TSQL.Statements.Parsers.ITSQLStatementParser.Parse()
at TSQL.TSQLStatementReader.MoveNext()
at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
at TSQL.TSQLStatementReader.ParseStatements(String tsqlText, Boolean useQuotedIdentifiers, Boolean includeWhitespace)
at Testing.App.SQLMethods.ParseSQLStatements(String sqlText) in D:\Testing.App\Testing.App\SQLMethods.cs:line 72
```

Kindly help me out if I am missing something or that's the expected behavior. This is how I'm doing:

```
string sqlText = @"SELECT system_user;";

var statements = TSQLStatementReader.ParseStatements(sqlText);

```

I am using nuget package version: 2.1.0

interestingly if I change the select statement to this

`SELECT system_user from tbl;`

I just added a FROM clause and it works, it gives me the statement and tokens.

**Comment by bruce-dunwiddie** [comment, marked as answer]
Thank you for bringing this up. I've created an issue from it, https://github.com/bruce-dunwiddie/tsql-parser/issues/93 .

I think the problem is stemming from an intermix of system "variables" and system "functions". In the previous major version, the differentiation didn't matter, because there wasn't any context aware parsing. With the new context aware parsing, I think it's considering this a function, and trying to parse arguments, but also it shouldn't even matter because it should bail out at the end of the string, which it's also not.

So please give me some time, a few days or so, to review and think on this one. It's likely to spawn fixes within multiple sections, instead of just fixing directly for this specific situation. In the meantime, you can pin to the last release of the previous major version v1.5.3, prior to the contextual parsing, and it likely won't have this issue.

**Reply by muneeebsheikh** [reply]
Okay I get it. 
Mean while I'll move to the previous release as you have mentioned.
Thank you for the suggestion.

**Reply by bruce-dunwiddie** [reply]
Should now be resolved in v2.2.0, https://www.nuget.org/packages/TSQL.Parser/2.2.0 .

Thank you again for making me aware. It was a simple change, that had wider impact than your simple reproduction, so it was good to get resolved.
